packages/components/organisms/f-form/src/components/FormField.vue

```diff
@@ -0,0 +1,150 @@
+<template>
+    <form-field
+        v-bind="fieldProps"
+        @input="updateField({ fieldName: fieldData.name, value: $event })"
+        @blur="hasInvalidErrorMessage && formFieldBlur()">
+        <template
+            v-if="hasError"
+            #error>
+            <error-message
+                :id="translations.errorName"
+                data-js-error-message
+                :data-test-id="translations[`${errorType}Error`]">
+                {{ translations.validationMessages[errorType] }}
+            </error-message>
+        </template>
+    </form-field>
+</template>
+
+<script>
+import FormField from '@justeat/f-form-field';
+import ErrorMessage from '@justeat/f-error-message';
+import { FORM_EVENTS } from '../constants';
+
+export default {
+    components: {
+        ErrorMessage,
+        FormField
+    },
+
+    props: {
+        fieldData: {
+            type: Object,
+            required: true
+        }
+    },
+
+    inject: ['$v'],
+
+    computed: {
```

Comment:
It feels to me that a lot of the logic here can be _stored_ in a separate place

Reply:
I've moved everything around a fair bit now. Hopefully it makes sense 😬